lib/services/elasticsearch.js

```diff
@@ -153,519 +166,479 @@ class ElasticSearch extends Service {
+          author: String(userId),
           createdAt: Date.now(),
           updatedAt: Date.now(),
-          updater: userId,
-          active: true,
-          deletedAt: null
+          updater: String(userId)
         };
-        // extends the response with the source from request
-        // When we write in ES, the response doesn't contain the initial document content
-        return this.client.exists(existQuery);
+
+        return this.client.exists({ index: esIndex, id });
       })
-      .then(exist => {
-        if (exist) {
-          return this.client.index(esRequest);
+      .then(({ body: exists }) => {
+        if (! exists) {
+          return this.reject('document_not_found', id);
         }
 
-        return Bluebird.reject(this.getError(
-          'document_not_found',
-          esRequest.id
-        ));
+        return this.client.index(esRequest);
       })
-      .then(result => this.refreshIndexIfNeeded(
-        esRequest,
-        _.extend(result, {_source: request.input.body}))
-      );
+      .then(({ body }) => {
+        const apiResponse = {
+          _id: id,
+          _version: body._version,
+          _source: esRequest.body
+        };
+
+        return apiResponse;
+      });
   }
 
   /**
    * Send to elasticsearch the document id to delete
    *
-   * @param {Request} request
-   * @returns {Promise} resolve an object that contains _id
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {string} id - Document id
+   * @param {object} options - refresh (null)
+   *
+   * @returns {Promise} { _id }
    */
-  delete(request) {
-    const esRequest = initESRequest(request, ['refresh', 'retryOnConflict']);
-
-    esRequest.id = request.input.resource._id;
+  delete (index, collection, id, { refresh=null }) {
+    const esRequest = {
+      index: this._buildClient._getESIndex(index, collection),
+      id,
+      refresh,
+      retryOnConflict
+    };
 
     assertWellFormedRefresh(esRequest);
 
-    return this.client
-      .get({
-        index: esRequest.index,
-        type: esRequest.type,
-        id: esRequest.id
-      })
-      .then(result => {
-        if (result._source._kuzzle_info && !result._source._kuzzle_info.active) {
-          this.throw('document_not_found', esRequest.id);
-        }
-        // injecting retryOnConflict default configuration
-        if (
-          !_.has(esRequest, 'retryOnConflict')
-          && this.config.defaults.onUpdateConflictRetries > 0
-        ) {
-          esRequest.retryOnConflict = this.config.defaults.onUpdateConflictRetries;
-        }
-
-        // Add metadata
-        esRequest.body = {
-          doc: {
-            _kuzzle_info: {
-              active: false,
-              deletedAt: Date.now(),
-              updater: this._getUserId(request)
-            }
-          }
+    return this.client.delete(esRequest)
+      .then(() => {
+        const apiResponse = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "apiResponse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)